src/rest/connector/libs/xpresso/implementation.py

```diff
@@ -0,0 +1,310 @@
+    for xpresso
+
+    YAML Example
+    ------------
+
+        devices:
+            xpresso:
+                os: xpresso
+                connections:
+                    rest:
+                        class: rest.connector.Rest
+                        host : "xpresso.cisco.com"
+                        credentials:
+                            rest:
+                                token: <xpressoaccesstoken>
+
+    Code Example
+    ------------
+
+        >>> from pyats.topology import loader
+        >>> testbed = loader.load('testbed.yaml')
+        >>> device = testbed.devices['xpresso']
+        >>> device.connect(alias='rest', via='rest')
+        >>> device.rest.connected
+        True
+    '''
+    @BaseConnection.locked
+    def connect(self, timeout=30):
+        '''connect to the device via REST
+
+        Arguments
+        ---------
+
+            timeout (int): Timeout value
+
+        Raises
+        ------
+
+        Exception
+        ---------
+
+            If the connection did not go well
+
+        Note
+        ----
+
+        There is no return from this method. If something goes wrong, an
+        exception will be raised.
+
+
+        YAML Example
+        ------------
+        devices:
+            xpresso:
+                os: xpresso
+                connections:
+                    rest:
+                        class: rest.connector.Rest
+                        host : "xpresso.cisco.com"
+                        protocol: http
+                        credentials:
+                            rest:
+                                token: <xpressoaccesstoken>
+
+        Code Example
+        ------------
+
+            >>> from pyats.topology import loader
+            >>> testbed = loader.load('testbed.yaml')
+            >>> device = testbed.devices['xpresso']
+            >>> device.connect(alias='rest', via='rest')
+        '''
+
+        if self.connected:
+            return True
+
+        if 'host' in self.connection_info:
+            host = self.connection_info['host']
+        else:
+            raise Exception("must include a host name")
+
+        if 'protocol' in self.connection_info:
+            protocol = self.connection_info['protocol']
+        else:
+            protocol = 'https'
+            
+        if 'port' in self.connection_info:
+            port = self.connection_info['port']
+            self.url = f'{protocol}://{host}:{port}/'
+        else:
+            self.url = f'{protocol}://{host}/'
+
+        # get authentication data
+        token = get_token(self)
+        if token is None:
+            raise Exception("must include an access token for rest")
+
+        # start sesssion
+        log.info(f"Connecting to {self.device.name} with alias {self.alias}")
+        self.session = requests.Session()
+        self.session.trust_env = False
+        self.session.headers.update({"Authorization": token})
+        self._is_connected = True
```

Comment:
shouldn't we a request.get here to make sure that we get response ok and then set self._connect to True?

Reply:
For xpresso there isn't really a connection do a device/platform, so not sure if there is a point in doing this.